src/models/krate.rs

```diff
@@ -427,14 +428,28 @@ impl Crate {
         match owner {
             // Users are invited and must accept before being added
             owner @ Owner::User(_) => {
-                insert_into(crate_owner_invitations::table)
+                let maybe_inserted = insert_into(crate_owner_invitations::table)
                     .values(&NewCrateOwnerInvitation {
                         invited_user_id: owner.id(),
                         invited_by_user_id: req_user.id,
                         crate_id: self.id,
                     })
                     .on_conflict_do_nothing()
-                    .execute(conn)?;
+                    .get_result::<CrateOwnerInvitation>(conn)
+                    .optional()?;
+
+                if maybe_inserted.is_some() {
+                    if let Owner::User(user) = &owner {
```

Comment:
Because we're in [this arm of the match](https://github.com/rust-lang/crates.io/pull/1909/commits/8838932b407dad8bf51bc8e0807e6bcdbcaa40f6#diff-1307fe21b99862533118a71d3c111feaR430), we know this `if let` will always be true. We should be able to:

- Remove this line
- Change the pattern for this arm to `Owner::User(user)` (I thought we could get both with `owner @ Owner::User(user)` but I just tried it and that doesn't work, but we don't actually need both, keep reading)
- Change [this line](https://github.com/rust-lang/crates.io/pull/1909/commits/8838932b407dad8bf51bc8e0807e6bcdbcaa40f6#diff-1307fe21b99862533118a71d3c111feaR433) to `invited_user_id: user.id` because that's [what `owner.id()` does anyway](https://github.com/rust-lang/crates.io/blob/9042ebf469e18c068ddeadfbef842d482c87fa21/src/models/owner.rs#L80)
- Change [this line](https://github.com/rust-lang/crates.io/pull/1909/commits/8838932b407dad8bf51bc8e0807e6bcdbcaa40f6#diff-1307fe21b99862533118a71d3c111feaR455) to `user.gh_login`, because that's [what `owner.login()` does anyway](https://github.com/rust-lang/crates.io/blob/9042ebf469e18c068ddeadfbef842d482c87fa21/src/models/owner.rs#L73)

That'll let us get rid of a conditional that will always be true and the extra indentation that required. Could you make these changes please?

Reply:
Definitely! Yeah, I had attempted the match arm update you'd initially suggested in the issue description, and I think hit the same as you.

I made those updates and ran through a quick test, seems to be working as expected.